SonarQube.Client/HttpResponseException.cs

```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Net;
+
+namespace SonarQube.Client
+{
+    public class HttpResponseException : Exception
+    {
+        public HttpResponseException(HttpStatusCode statusCode, string reasonPhrase)
```

Comment:
Exceptions should be serializable (there's a snippet that correctly implements the pattern - just type `exception` then hit `tab`, `tab`)